### PR TITLE
feat(function): Support variant max/min functions

### LINKED ISSUE #5291: Feature: support max/min aggregation function for json value
**Summary**

support max/min aggregation function for json value
```sql
mysql> create table var (v variant);
mysql> insert into var select parse_json('{"a":"b"}');
mysql> insert into var select parse_json('[1,2,3]');
mysql> select max(v), min(v) from var;
ERROR 1105 (HY000): Code: 1002, displayText = Bincode does not support the serde::Deserializer::deserialize_any method (while in processor thread 0).
```

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support variant max/min functions

## Changelog

- New Feature
- Build/Testing/CI

## Related Issues

Fixes #5291

### CODE REVIEW

File: tests/suites/0_stateless/02_function/02_0000_function_aggregate_mix.sql [hunk 1]
```diff
@@ -23,4 +23,41 @@ SELECT std(number) between  2886.751 and 2886.752 from numbers_mt(10000);
...
-SELECT covar_pop(number, number) from (select * from numbers_mt(5) order by number asc);
\ No newline at end of file
+SELECT covar_pop(number, number) from (select * from numbers_mt(5) order by number asc);
+
+DROP DATABASE IF EXISTS db1;
+CREATE DATABASE db1;
+USE db1;
+
+SELECT '==Variant==';
+CREATE TABLE IF NOT EXISTS t1(id Int null, var Variant null) Engine = Memory;
```
Comment (BohuTANG) [inline tests/suites/0_stateless/02_function/02_0000_function_aggregate_mix.sql:33]: I prefer to remove the `Engine = Memory`. Let's use `fuse` engine by default for all the tests.

Comment (b41sh) [reply tests/suites/0_stateless/02_function/02_0000_function_aggregate_mix.sql:33]: ok

File: common/functions/src/aggregates/aggregate_arg_min_max.rs [hunk 2]
```diff
@@ -298,25 +365,39 @@ pub fn try_create_aggregate_arg_minmax_function<const IS_MIN: bool>(
...
-        if IS_MIN {
-            type State = ArgMinMaxState<$T, CmpMin>;
-             AggregateArgMinMaxFunction::<$T, CmpMin, State>::try_create(
-                display_name,
-                arguments,
-            )
+    let phid = data_type.data_type_id().to_physical_type();
+
+    with_match_scalar_type!(phid, |$T| {
+        if phid == PhysicalTypeID::Variant {
```
Comment (fkuner) [inline common/functions/src/aggregates/aggregate_arg_min_max.rs:371]: `with_match_scalar_type` seems to match `PhysicalTypeID::Variant`

Comment (fkuner) [reply common/functions/src/aggregates/aggregate_arg_min_max.rs:371]: Oh, the state is different.

**Comment by fkuner** [comment]
LGTM

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@soyeric128 This PR needs document, thanks.

**Comment by soyeric128** [comment]
> @soyeric128 This PR needs document, thanks.
> https://github.com/datafuselabs/databend/pull/5525/files#diff-948dbda02a352e81a02bc2b8508c4d735bef68f86c270d5e7b7f6f8f229020e9

Hi @b41sh and @BohuTANG ,

I found the documentation already includes the functions arg_max and arg_min, but it does not mention that the **_val_** argument supports the Variant type. 
So for this feature, I would update both functions by adding the information about all the supported types of the **_val_** argument. Could you pls help confirm which data types the **_val_** argument can support currently, besides Variant? Thanks.

[https://databend.rs/doc/reference/functions/aggregate-functions/aggregate-argmax](https://databend.rs/doc/reference/functions/aggregate-functions/aggregate-argmax)
[https://databend.rs/doc/reference/functions/aggregate-functions/aggregate-argmin](https://databend.rs/doc/reference/functions/aggregate-functions/aggregate-argmin)

**Comment by BohuTANG** [comment]
Thank you, @soyeric128 .
cc @sundy-li

**Comment by sundy-li** [comment]
arg_min( arg, val)

The first arg supports all data types.
The second arg supports all data types except `Tuple` (`Tuple` is not stable now as it does not mention in https://databend.rs/doc/reference/data-types/)

**Comment by soyeric128** [comment]
Thank you @sundy-li . So currently we can say both **arg** and **val** support all the data types mentioned at [https://databend.rs/doc/reference/data-types/](https://databend.rs/doc/reference/data-types/), right?

**Pull request merged**